proposals/proposal-0001.md

```diff
@@ -0,0 +1,223 @@
+   including the proposal number, a short descriptive title, the author names,
+   etc.
+1. Abstract - a short (~200 word) description of the technical issue being
+   addressed.
+1. Specification - The technical specification should describe the syntax and
+   semantics of any new feature. If the proposal describes a standard, the
+   specification should be detailed enough to describe how other community
+   members may re-implement the standard in their product.
+1. Motivation - The motivation is critical for proposals that want to change
+   Helm in a significant way. It should clearly explain why existing features
+   are inadequate to address the problem that the proposal solves. proposal
+   submissions without sufficient motivation may be rejected outright.
+1. Rationale - The rationale fleshes out the specification by describing what
+   motivated the design and why particular design decisions were made. It should
+   describe alternate designs that were considered and related work, e.g. how
+   the feature is supported in other projects.
+1. Backwards Compatibility - All proposals that introduce backwards
+   incompatibilities must include a section describing these incompatibilities
+   and their severity. The proposal must explain how the author proposes to deal
+   with these incompatibilities. proposal submissions without a sufficient
+   backwards compatibility treatise may be rejected outright.
+1. Reference Implementation - The reference implementation must be completed
+   before any proposal is given status "Final", but it need not be completed
+   before the proposal is accepted. While there is merit to the approach of
+   reaching consensus on the specification and rationale before writing code,
+   the principle of "rough consensus and running code" is still useful when it
+   comes to resolving many discussions of API details.
+
+The final implementation must include test code and documentation appropriate
+for the Helm project's reference or the community at large.
+
+## Proposal header preamble
+
+Each proposal must begin with a [YAML][yaml] style header preamble. The headers
+must appear in the following order:
+
+Headers marked with "*" are optional and are described below. All other headers
+are required.
+
+```yaml
+---
+proposal: <proposal number>
+title: "<proposal title>"
+authors: [ "<list of authors' real names and optionally, email addresses>" ]
+created: 2020-08-13
+* helm-version: "<version number>"
+* requires: [ <proposal numbers> ]
+* replaces: <proposal number>
+* superseded-by: <proposal number>
+---
+```
+
+The `authors` header lists the names, and optionally the email addresses of all
+the authors/owners of the proposal. The format of the author header value must
+be
+
+```yaml
+authors: [ "John Doe <john@example.com>", "Jane Doe <jane@example.com>" ]
+```
+
+If the email address is included, and
+
+```yaml
+authors: [ "John Doe", "Jane Doe" ]
```

Comment:
Should they be names or GitHub handles?

Reply:
Given that real names can have namespace collisions, perhaps GitHub handles are slightly less ambiguous? (She says, in a thread where two Matts are discussing name disambiguation.)

Reply:
After deliberating over handles vs. names, I still believe legal names with an optional email is the right decision here. My goal here is to ensure that the HIP system can be used as reference material for the next few decades. And in 20+ years, I am concerned that technologies like github may no longer exist, or we may switch to a new system over time. Email was first introduced in the 60's and isn't going anywhere any time soon. :)

For a frame of reference, the first PEP was written in 2000, and is still being used as reference material today. Github was founded in 2008. And while I didn't state them earlier, IETF RFCs were written back in 1969 using legal names in the `Author` field.

My other concern is that the "@" format is not clear enough to the reader where the handle originated from. @bacongobbler could refer to my twitter, github, instagram, gitlab, slack, my discord, or any other new-and-trending social media platform in the next 20+ years (TikTok?).

One other example: say that in 20 years, these HIPs become published in a book. Anything in the preamble is likely to be displayed in a document as a subtitle or metadata, likely in the form `Author: @bacongobbler`. In 20 years, if I provided you with a book that had `Author: @deis`, could you tell me whether it was [deis-the-company](https://github.com/deis), or [Matt Greene](https://twitter.com/deis) who wrote the proposal?

Reply:
Good analysis @bacongobbler. name and optional email sounds good.